internal/gengapic/testdata/rest_ServerStreamRPC.want

```diff
@@ -0,0 +1,90 @@
+func (c *fooRESTClient) ServerStreamRPC(ctx context.Context, req *foopb.Foo, opts ...gax.CallOption) (foopb.FooService_ServerStreamRPCClient, error) {
+	m := protojson.MarshalOptions{AllowPartial: true}
+	jsonReq, err := m.Marshal(req)
+	if err != nil {
+		return nil, err
+	}
+
+	baseUrl, _ := url.Parse(c.endpoint)
+	baseUrl.Path += fmt.Sprintf("/v1/foo")
+
+	// Build HTTP headers from client and context metadata.
+	headers := buildHeaders(ctx, c.xGoogMetadata, metadata.Pairs("Content-Type", "application/json"))
+	var streamClient *serverStreamRPCRESTClient
+	e := gax.Invoke(ctx, func(ctx context.Context, settings gax.CallSettings) error {
+		httpReq, err := http.NewRequest("POST", baseUrl.String(), bytes.NewReader(jsonReq))
+		if err != nil {
+			return err
+		}
+		httpReq = httpReq.WithContext(ctx)
+		httpReq.Header = headers
+
+		httpRsp, err := c.httpClient.Do(httpReq)
+		if err != nil{
+			return err
+		}
+
+		if err = googleapi.CheckResponse(httpRsp); err != nil {
+			return err
+		}
+
+		streamClient = &serverStreamRPCRESTClient{
+			ctx: ctx,
+			md: metadata.MD(httpRsp.Header),
+			stream: gax.NewProtoJSONStreamReader(httpRsp.Body, (&foopb.Foo{}).ProtoReflect().Type()),
+		}
+		return nil
+	}, opts...)
+
+	return streamClient, e
+}
+
+// serverStreamRPCRESTClient is the stream client used to consume the server stream created by
+// the REST implementation of ServerStreamRPC.
+type serverStreamRPCRESTClient struct {
+	ctx context.Context
+	md metadata.MD
+	stream *gax.ProtoJSONStream
+}
+
+func (c *serverStreamRPCRESTClient) Recv() (*foopb.Foo, error) {
+	if err := c.ctx.Err(); err != nil {
+		defer c.stream.Close()
+		return nil, err
+	}
+	msg, err := c.stream.Recv()
+	if err != nil {
+		defer c.stream.Close()
+		return nil, err
+	}
+	res := msg.(*foopb.Foo)
+	return res, nil
+}
+
+func (c *serverStreamRPCRESTClient) Header() (metadata.MD, error) {
+	return c.md, nil
+}
+
+func (c *serverStreamRPCRESTClient) Trailer() metadata.MD {
+	return c.md
+}
+
+func (c *serverStreamRPCRESTClient) CloseSend() error {
+	// This is a no-op to fulfill the interface.
+	return nil
+}
+
+func (c *serverStreamRPCRESTClient) Context() context.Context {
+	return c.ctx
+}
+
+func (c *serverStreamRPCRESTClient) SendMsg(m interface{}) error {
```

Comment:
I wonder if these "noop" impls should return errors when they are not supported, thoughts?

Reply:
Yeah, I think that would be fine! Is it worth making a single "ErrUnimplemented" error? WDYT is best?

Reply:
I am not sure we need a sentinel, I don't think it provides too much value at runtime to know it was an unimplmented error as this will only be run into at dev time. Maybe just return a grpc.Status with a code unimiplmented?

Reply:
Done!